### ISSUE TITLE
Setup for authorization with Bearer token

### ISSUE THREAD

**Issue by marcelino-borges** [davibaltar/swagger-autogen#190]
Hi there, it might be a silly question, but allow me...

Right now I have this:

```
const SWAGGER_OUTPUT_PATH = "./swagger_output.json";
  const ROUTES_PATH = ["./src/routes/index.ts"];
  const apiDescription = (await getPackageJson()).description;

  const doc = {
    info: {
      version: apiVersion,
      title: apiDescription.description,
    },
    servers: [
      {
        url: "http://localhost:3000/",
        description: "Local server",
      },
      {
        url: "https://users-api-dev.onrender.com/",
        description: "On Render [DEV]",
      },
    ],
    basePath: "/",
    schemes: ["http", "https"],
    consumes: ["application/json"],
    produces: ["application/json"],
    definitions: {
      Error: resultError,
      SuccessMessage: resultSuccessMessage,
      CreateUserProfile: CreateProfile,
      UserProfile: UpdateProfile,
    },
    securityDefinitions: {
      bearerAuth: {
        type: "oauth2",
        bearerFormat: "JWT",
        in: "header",
        flow: "implicit",
        name: "Authorization",
        description: "Provide a JWT bearer token",
        value: "Bearer XXXXXXXXXXXXXXXXXX",
        scopes: {
          all: "Applied to all requests",
        },
      },
    },
  };

  swaggerAutogen()(SWAGGER_OUTPUT_PATH, ROUTES_PATH, doc);
```

Everything works fine, but my authorization.
- I have a middleware validating a Bearer token against my endpoints
- I thave this instruction in my protected endpoints:
```
    #swagger.security = [{
      "bearerAuth": ["all"]
    }]
```
- My authorization dialog looks like this:
![image](https://user-images.githubusercontent.com/33501037/216778842-1cf9356d-f1ce-4811-bfdf-3969806f7375.png)

But I have this field showing up in my endpoints:
![image](https://user-images.githubusercontent.com/33501037/216778873-f18f735b-16dc-4e2e-a98f-98cff1e45cba.png)

What am I doing wrong?

Thanks in advance!

**Comment by davibaltar** [comment]
Hi @marcelino-borges! Thank you for opening the issue. 

For now, we don't have a way to disable/hide a specific parameter. But a workaround that you can do is use the tag:

```js
// #swagger.autoHeaders=false
```

to disable the automatic headers recognition. NOTE! This tag is available in version >= 2.23.0 and it disables for all header parameters in the function. If it doesn't work there, let me know. 

But I'll mark this issue for a future implementations about this, for the possibility to disable/hide for a specific parameter. Thanks and any news I'll post here.

**Comment by marcelino-borges** [comment]
Got it.
But in this case that swagger handles the authorization in the that dialog (where we set the access token for the whole API), should we really see this header input in the endpoints?
Considering we set the securityDefinitions etc...

**Comment by davibaltar** [comment]
Indeed. I'll try to handle this as well. If the endpoint has the tag `#swagger.security=...` and has match between the securityDefinitions.(...).name and the header parameter name in the function, the header input won't be written in the .json file.
